riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/RateLimitResetDelayFunction.java

```diff
@@ -31,13 +32,14 @@ public RateLimitResetDelayFunction(final Clock clock) {
     }
 
     @Override
-    public Duration computeDelay(final ClientHttpResponse result, final Throwable failure, final ExecutionContext context) {
-        return Optional.ofNullable(failure)
+    public Duration get(final ExecutionContext<ClientHttpResponse> context) {
+        return ofNullable(context)
+                .map(ExecutionContext::getLastException)
                 .filter(HttpResponseException.class::isInstance)
                 .map(HttpResponseException.class::cast)
                 .map(response -> response.getResponseHeaders().getFirst("X-RateLimit-Reset"))
                 .map(parser::parse)
-                .orElse(null);
+                .orElse(Duration.ofMinutes(-1));
```

Comment:
Changed to prevent NPE in (DelayablePolicy)[https://github.com/failsafe-lib/failsafe/blob/master/core/src/main/java/dev/failsafe/spi/DelayablePolicy.java#L52], behaviour for negative result and null result is the same at the end - see  (DelayablePolicy)[https://github.com/failsafe-lib/failsafe/blob/master/core/src/main/java/dev/failsafe/spi/DelayablePolicy.java#L62]